libduckdb-sys/openssl/find_normal.rs

```diff
@@ -0,0 +1,274 @@
+and include information about your system as well as this message.
+
+$HOST = {}
+$TARGET = {}
+openssl-sys = {}
+
+",
+        host,
+        target,
+        env!("CARGO_PKG_VERSION")
+    );
+
+    if host.contains("apple-darwin") && target.contains("apple-darwin") {
+        let system = Path::new("/usr/lib/libssl.0.9.8.dylib");
+        if system.exists() {
+            msg.push_str(
+                "
+
+openssl-sys crate build failed: no supported version of OpenSSL found.
+
+Ways to fix it:
+- Use the `vendored` feature of openssl-sys crate to build OpenSSL from source.
+- Use Homebrew to install the `openssl` package.
+
+",
+            );
+        }
+    }
+
+    if host.contains("unknown-linux")
+        && target.contains("unknown-linux-gnu")
+        && Command::new("pkg-config").output().is_err()
+    {
+        msg.push_str(
+            "
+It looks like you're compiling on Linux and also targeting Linux. Currently this
+requires the `pkg-config` utility to find OpenSSL but unfortunately `pkg-config`
+could not be found. If you have OpenSSL installed you can likely fix this by
+installing `pkg-config`.
+
+",
+        );
+    }
+
+    if host.contains("windows") && target.contains("windows-gnu") {
+        msg.push_str(
+            "
+It looks like you're compiling for MinGW but you may not have either OpenSSL or
+pkg-config installed. You can install these two dependencies with:
+
+pacman -S openssl-devel pkg-config
+
+and try building this crate again.
+
+",
+        );
+    }
+
+    if host.contains("windows") && target.contains("windows-msvc") {
+        msg.push_str(
+            "
+It looks like you're compiling for MSVC but we couldn't detect an OpenSSL
+installation. If there isn't one installed then you can try the rust-openssl
+README for more information about how to download precompiled binaries of
+OpenSSL:
+
+https://github.com/sfackler/rust-openssl#windows
+
+",
+        );
+    }
+
+    panic!("{}", msg);
+}
+
+/// Attempt to find OpenSSL through pkg-config.
+///
+/// Note that if this succeeds then the function does not return as pkg-config
+/// typically tells us all the information that we need.
+fn try_pkg_config() -> Result<(), ()> {
+    let target = env::var("TARGET").unwrap();
+    let host = env::var("HOST").unwrap();
+
+    // If we're going to windows-gnu we can use pkg-config, but only so long as
+    // we're coming from a windows host.
+    //
+    // Otherwise if we're going to windows we probably can't use pkg-config.
+    if target.contains("windows-gnu") && host.contains("windows") {
+        env::set_var("PKG_CONFIG_ALLOW_CROSS", "1");
+    } else if target.contains("windows") {
+        return Ok(());
+    }
+
+    let lib = match pkg_config::Config::new().print_system_libs(false).probe("openssl") {
+        Ok(lib) => lib,
+        Err(e) => {
+            println!("run pkg_config fail: {:?}", e);
+            return Ok(());
+        }
+    };
+
+    super::postprocess(&lib.include_paths);
```

Comment:
I think I get it. It's because the function returns nothing but callback to the postprocess function. Which means in our build.rs the `if let` will not work.

Whether we also pass the value back using this postprocess, or we return the path back to the function

Reply:
They will exit because the save the included file path in the postprocess function. Yes we need the include_dirs so the compiler can find the header file.

through err may not work, should through Ok

Reply:
You can go through the find ssl to understand the whole logic

Reply:
You can also grant your repo permission to me if you want me to help fix the issue

Reply:
That makes sense, although the `Err()` here is not a true error, it's just a trick to short-circuit the execution logic (basically take advantage of the `?` operator). I would suggest that we change the return type of `Err()` to also include the paths, and then return an `Err(...)` here.

Reply:
Also just granted you access. Would love your help!

Reply:
If return err we need to handle the err carefully. Yes anyway we need to find way to return the include_dirs

Reply:
We just need to return the include path in the error (instead of a unit) and extend the match statement in the main code to handle the error case too! I think it should be very straightforward